### PR TITLE
Add error for multiple null suppressions

### LINKED ISSUE #37111: Block `expr!!` (double suppression)
We may want to make `!!` a token in the parameter null-checking checking some time after C# 8.

### PR DESCRIPTION
**Pull request by agocke**
Fixes #37111

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -52250,7 +52238,59 @@ static void G(string s)
...
-            comp.VerifyDiagnostics();
+            comp.VerifyDiagnostics(
+                // (5,11): error CS8715: Duplicate null suppression operator ('!')
+                //         G(default!!);
+                Diagnostic(ErrorCode.ERR_DuplicateNullSuppression, "default").WithLocation(5, 11),
+                // (6,11): error CS8715: Duplicate null suppression operator ('!')
+                //         G(s!!);
+                Diagnostic(ErrorCode.ERR_DuplicateNullSuppression, "s").WithLocation(6, 11),
+                // (7,12): error CS8715: Duplicate null suppression operator ('!')
+                //         G((s!)!);
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >(s!)! [](start = 29, length = 5)

Does this need to be an error?

Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: This is pretty much just reserving the right to consider `!!` a distinct operator in the future, right? So probably no need to complain in this scenario.

Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: I expected we would implement this as a parsing error: `expr!!` would be an error, but `(expr!)!` would not.
I wouldn't push back very hard on this given the timing though, so up to you.

---
In reply to: [302638675](https://github.com/dotnet/roslyn/pull/37139#discussion_r302638675) [](ancestors = 302638675)

**Comment by agocke** [comment]
I think this should be a semantic error for now. We can always relax it later.

**Pull request merged**